app/grandchallenge/algorithms/views.py

```diff
@@ -593,21 +607,63 @@ def columns(self):
             Column(title="Viewer", sort_field="inputs__image__files__file"),
         ]
 
-        if "PDF" in self.output_interface_kinds:
-            columns.append(
-                Column(title="PDF", sort_field="", classes=("nonSortable",)),
-            )
-
-        if "CHART" in self.output_interface_kinds:
-            columns.append(
-                Column(title="Chart", sort_field="", classes=("nonSortable",)),
-            )
+        for grouped_interfaces in self.outputs_list_display.values():
+            for interface in grouped_interfaces:
+                if interface.kind in (
+                    InterfaceKind.InterfaceKindChoices.STRING,
+                    InterfaceKind.InterfaceKindChoices.INTEGER,
+                    InterfaceKind.InterfaceKindChoices.FLOAT,
+                    InterfaceKind.InterfaceKindChoices.BOOL,
+                ):
```

Comment:
Nitpick: Rather than having the same list of things here what you could do is take the key in L610 by looping over the items (`for key, grouped_interfaces in self.outputs_list_display.items():`) and then checking `if key == "JSON"`. Makes it easier for a future developer to change in the future.